### ISSUE TITLE
how to give the not tfrecords data input,when i you slim.learning.train!

### ISSUE THREAD

**Issue by andongchen** [tensorflow/models#1282]
https://github.com/tensorflow/models/blob/master/slim/train_image_classifier.py
when i use this model,but my new data is not tfrecords,and can't change to tfrecod, because my new data is a pair of data,i use to person re-identification !in this case,hou can i input my new data in this function,my method is like this,but it like only use once my generate data function.because in this way the model converge very fast.

`
train_images, train_labels, train_labels_retrival = generate_batch_data.generate_batch_data(
        batch_size=FLAGS.batch_size,
        split_name='photo')

    batch_queue = slim.prefetch_queue.prefetch_queue(
        [train_images, train_labels, train_labels_retrival], capacity=2 * deploy_config.num_clones)

    ####################
    # Define the model #
    ####################

    def clone_fn(batch_queue):
      """Allows data parallelism by creating multiple clones of network_fn."""
      train_images, train_labels, train_labels_retrival = batch_queue.dequeue()
      logits, end_points = network_fn(train_images)
      correct = tf.cast(tf.equal(tf.argmax(tf.nn.softmax(logits),1), tf.argmax(slim.one_hot_encoding(train_labels, 751),1)), tf.float32)
      accuracy = tf.reduce_mean(correct)
      tf.summary.scalar('accuracy_identificaiton', accuracy)

      with tf.variable_scope('siamese'):
          f_1 = tf.slice(end_points['pool5'],[0, 0, 0, 0], [FLAGS.batch_size, -1, -1, -1])
          f_2 = tf.slice(end_points['pool5'], [FLAGS.batch_size, 0, 0, 0], [-1, -1, -1, -1])
          eucd2 = tf.pow(tf.subtract(f_1, f_2),2)
          # net = tf.reduce_mean(eucd2, [1, 2], name='pool1', keep_dims=True)
          net = slim.conv2d(eucd2, 2, [1, 1], activation_fn=None,
                                 normalizer_fn=None, scope='siamese')
          y_ = tf.squeeze(net, [1, 2], name='SpatialSqueeze')

      correct_verification = tf.cast(tf.equal(tf.argmax(tf.nn.softmax(y_), 1), tf.argmax(train_labels_retrival, 1)), tf.float32)
      accuracy_verification = tf.reduce_mean(correct_verification)
      tf.summary.scalar('accuracy_verification', accuracy_verification)

      #############################
      # Specify the loss function #
      #############################
      tf.losses.softmax_cross_entropy(
           logits=logits,onehot_labels=slim.one_hot_encoding(train_labels, 751),
          label_smoothing=FLAGS.label_smoothing, weights=0.5,scope='identification_loss')

      tf.losses.softmax_cross_entropy(logits=y_, onehot_labels=train_labels_retrival ,weights=1.0,scope='verification_loss')

      return end_points

    # Gather initial summaries.
    summaries = set(tf.get_collection(tf.GraphKeys.SUMMARIES))

    clones = model_deploy.create_clones(deploy_config, clone_fn,[batch_queue])
    first_clone_scope = deploy_config.clone_scope(0)
    # Gather update_ops from the first clone. These contain, for example,
    # the updates for the batch_norm variables created by network_fn.
    update_ops = tf.get_collection(tf.GraphKeys.UPDATE_OPS, first_clone_scope)

    # Add summaries for end_points.
    end_points = clones[0].outputs
    for end_point in end_points:
      x = end_points[end_point]
      summaries.add(tf.summary.histogram('activations/' + end_point, x))
      summaries.add(tf.summary.scalar('sparsity/' + end_point,
                                      tf.nn.zero_fraction(x)))

    # Add summaries for losses.
    for loss in tf.get_collection(tf.GraphKeys.LOSSES, first_clone_scope):
      summaries.add(tf.summary.scalar('losses/%s' % loss.op.name, loss))

    # Add summaries for variables.
    for variable in slim.get_model_variables():
      summaries.add(tf.summary.histogram(variable.op.name, variable))

    #################################
    # Configure the moving averages #
    #################################
    if FLAGS.moving_average_decay:
      moving_average_variables = slim.get_model_variables()
      variable_averages = tf.train.ExponentialMovingAverage(
          FLAGS.moving_average_decay, global_step)
    else:
      moving_average_variables, variable_averages = None, None

    #########################################
    # Configure the optimization procedure. #
    #########################################
    with tf.device(deploy_config.optimizer_device()):
      learning_rate = _configure_learning_rate(10936, global_step)
      optimizer = _configure_optimizer(learning_rate)
      summaries.add(tf.summary.scalar('learning_rate', learning_rate))

    if FLAGS.sync_replicas:
      # If sync_replicas is enabled, the averaging will be done in the chief
      # queue runner.
      optimizer = tf.train.SyncReplicasOptimizer(
          opt=optimizer,
          replicas_to_aggregate=FLAGS.replicas_to_aggregate,
          variable_averages=variable_averages,
          variables_to_average=moving_average_variables,
          replica_id=tf.constant(FLAGS.task, tf.int32, shape=()),
          total_num_replicas=FLAGS.worker_replicas)
    elif FLAGS.moving_average_decay:
      # Update ops executed locally by trainer.
      update_ops.append(variable_averages.apply(moving_average_variables))

    # Variables to train.
    variables_to_train = _get_variables_to_train()

    #  and returns a train_tensor and summary_op
    total_loss, clones_gradients = model_deploy.optimize_clones(
        clones,
        optimizer,
        var_list=variables_to_train)
    # Add total_loss to summary.
    summaries.add(tf.summary.scalar('total_loss', total_loss))

    # Create gradient updates.
    grad_updates = optimizer.apply_gradients(clones_gradients,
                                             global_step=global_step)
    update_ops.append(grad_updates)

    update_op = tf.group(*update_ops)
    train_tensor = control_flow_ops.with_dependencies([update_op], total_loss,
                                                      name='train_op')

    # Add the summaries from the first clone. These contain the summaries
    # created by model_fn and either optimize_clones() or _gather_clone_loss().
    summaries |= set(tf.get_collection(tf.GraphKeys.SUMMARIES,
                                       first_clone_scope))

    # Merge all summaries together.
    summary_op = tf.summary.merge(list(summaries), name='summary_op')

    ###########################
    # Kicks off the training. #
    ###########################
    slim.learning.train(
        train_tensor,
        logdir=FLAGS.train_dir,
        master=FLAGS.master,
        is_chief=(FLAGS.task == 0),
        init_fn=_get_init_fn(),
        summary_op=summary_op,
        number_of_steps=FLAGS.max_number_of_steps,
        log_every_n_steps=FLAGS.log_every_n_steps,
        save_summaries_secs=FLAGS.save_summaries_secs,
        save_interval_secs=FLAGS.save_interval_secs,
        sync_optimizer=optimizer if FLAGS.sync_replicas else None)`

**Comment by aselle** [comment]
This question is better asked on  [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a  bug or feature request. There is also a larger community that reads questions there. Thanks!
